02-intermediate-front-end/events-and-callbacks/readme.md

```diff
@@ -33,77 +76,592 @@ button.addEventListener("click", handleClickEvent)
+
+#### You do
+
+Find the keyCodes for:
+
+- Enter
+- Tab
+- Delete
+
+#### You do
+
+There are several other events that come up with the `input` tag. See if you can figure out the difference between:
+
+- `keyup`
+- `keydown`
+- `keypress`
+- `change`
+- `focus`
+- `blur`
+
+There are a bunch of different browser events you can use in Javascript, all [listed at W3Schools](http://www.w3schools.com/jsref/dom_obj_event.asp).
+
+> Yeah, yeah, I know, no-one likes W3Schools, but this list is accurate and easy-to-read.
+
+### &larr; &larr; &rarr; &rarr; &uarr; &darr; &uarr; &darr; a b &larrhk;
+
+There's a famous cheat code used in old Konami arcade games: you press up, up, down, down, left, right, left, right, 'b', 'a', and then 'start'. This code is so famous that many websites have secret functions for when users enter the Konami Code (with 'Enter' replacing 'start').
+
+Try it out at [Konami Code Sites](http://konamicodesites.com/).
+
+Some of those are now defunct. But my personal favorite is [Vogue.co.uk](http://www.vogue.co.uk/)
+
+This is possible on these websites thanks to Javascript event listeners.
+
+#### Turn and talk
+
+You've seen how to detect one keypress. **How might you make something happen after a specific series of keypresses?**
+
+## Event Defaults
+
+Back in the code we were using in-class, replace your button with a link to, say, Google:
+
+```html
+<body>
+  <a href="http://google.com">Click me!</a>
+</body>
+```
+
+Now, add an event listener to that link that brings up a `prompt` box, asking the user if they want to go to Google.
+
+```js
+var link = document.querySelector("a")
+var handleClickEvent = function(e){
+  var input = prompt("You sure you want to go to Google?")
+}
+link.addEventListener("click", handleClickEvent);
+```
+
+The problem is we don't know how to stop them from going to Google! They go anyway, whether they hit 'OK' or 'Cancel'.
+
+Some elements, like `<a>`, have a default action they do -- going to another webpage, in this case. You can prevent that default action with an Event property called `preventDefault`.
+
+```js
+var link = document.querySelector("a")
+var handleClickEvent = function(e){
+  e.preventDefault();
+  var input = prompt("You sure you want to go to Google?")
+}
+link.addEventListener("click", handleClickEvent);
+```
+
+Now, no matter what the user clicks, they won't go to Google.
+
+In order to make it so they that **do** go to Google on clicking OK, but **don't** on clicking 'Cancel', we can use the fact that when you click 'Cancel' on a `prompt`, it returns `null`:
+
+```js
+var button = document.querySelector("a")
+var handleClickEvent = function(e){
+  if(prompt("You sure you want to go to Google?") === null){
+    e.preventDefault();
+  }
+}
+button.addEventListener("click", handleClickEvent);
+```
+
+## Making something draggable and droppable with removeEventListener
```

Comment:
I don't know draggable/droppable is in the scope of this course? i'd rather leave it out. Seems like too much, this is a brand new thing for most, I think just teaching click and teaching it well will be of better service to us.